### DISCUSSION TITLE
QC ALERT

### DISCUSSION THREAD

**Discussion by drosop** [milaboratory/mixcr#1447] [Q&A] [answered]
Dear MiXCR team,

I have run the mouse BCR sequencing using Takara SMARTer® Mouse BCR IgG H/K/L Profiling Kit and did analysis using the MiXCR preset mixcr analyze takara-mouse-rna-bcr-smarter using MiXCR v4.5.0. Also, There was very low adaptor content, so I didnt do trimming before.

In the output, I have alerts for :

   Successfully aligned reads:                                     97.67% [OK]
   Off target (non TCR/IG) reads:                                 1.8%   [OK]
   Reads with no V or J hits:                                          0.49%  [OK]
   **Overlapped paired-end reads:                                  32.07% [ALERT]
   Reads used in clonotypes:                                         19.68% [ALERT]
   Alignments that do not cover VDJRegion:                68.03% [ALERT]**
   Alignments dropped due to low sequence quality:   0.0%   [OK]
   Clones dropped in post-filtering:                                0.0%   [OK]
   Alignments dropped in clones post-filtering:             0.0%   [OK]

However, the alignment rate is good. Is this report fine? What can be the reason for low number of reads used in clonotypes or VDJ regions? 

Thank you,

**Comment by MiLaboratoriesSupport** [comment]
​Hi,  
My guess is that you used a shorter reads' length (150+150?) which was not enough to cover the whole VDJRegion. By default MiXCR tries to assemble the full-length receptor sequence for this protocol. If that is the case, you may add `--assemble-clonotypes-by CDR3` parameter, to assemble clones by CDR3 region only.  ​   

Sincerely,
Mark

**Reply by mizraelson** [reply]
Hi, 
at the same time I see a high number of Overlapped paired-end reads: 32.07% for 150+150. Can you share the report files? 

Sincerely,
Mark

**Reply by drosop** [reply]
Hi,

The read length is 300 + 300.

Running:
mixcr align --report /rsrch6/scratch/lym_myl_rsch/plakra/SMARCA4/bcr/analysis/638_mBCR.align.report.txt --json-report /rsrch6/scratch/lym_myl_rsch/plakra/SMARCA4/bcr/analysis/638_mBCR.align.report.json --threads 16 --preset takara-mouse-rna-bcr-smarter --save-output-file-names /rsrch6/scratch/lym_myl_rsch/plakra/SMARCA4/bcr/analysis/638_mBCR.align.list --species mmu /rsrch6/scratch/lym_myl_rsch/plakra/SMARCA4/bcr/fastq/638_mBCR_R1.fastq.gz /rsrch6/scratch/lym_myl_rsch/plakra/SMARCA4/bcr/fastq/638_mBCR_R2.fastq.gz /rsrch6/scratch/lym_myl_rsch/plakra/SMARCA4/bcr/analysis/638_mBCR.vdjca
Alignment: 0%
Alignment: 11.7%  ETA: 00:01:00
Alignment: 23.1%  ETA: 00:00:40
Alignment: 35.5%  ETA: 00:00:26
Alignment: 45.7%  ETA: 00:00:21
Alignment: 56.9%  ETA: 00:00:15
Alignment: 68.3%  ETA: 00:00:13
Alignment: 79.5%  ETA: 00:00:11
Alignment: 91.9%  ETA: 00:00:03
====================== report: align ======================
Analysis time: 8.76m
Total sequencing reads: 799460
Successfully aligned reads: 738310 (92.35%)
Coverage (percent of successfully aligned):
  CDR3: 99.66%
  FR3_TO_FR4: 99.81%
  CDR2_TO_FR4: 99.84%
  FR2_TO_FR4: 80.76%
  CDR1_TO_FR4: 37.86%
  VDJRegion: 21.37%
Alignment failed: no hits (not TCR/IG?): 46225 (5.78%)
Alignment failed: absence of V hits: 11764 (1.47%)
Alignment failed: absence of J hits: 2918 (0.36%)
Alignment failed: no target with both V and J alignments: 243 (0.03%)
Overlapped: 213652 (26.72%)
Overlapped and aligned: 170731 (21.36%)
Overlapped and not aligned: 42921 (5.37%)
Alignment-aided overlaps, percent of overlapped and aligned: 0 (0%)
No CDR3 parts alignments, percent of successfully aligned: 54 (0.01%)
Partial aligned reads, percent of successfully aligned: 2483 (0.34%)
V gene chimeras: 5834 (0.73%)
J gene chimeras: 17 (0%)
Paired-end alignment conflicts eliminated: 80637 (10.09%)
Realigned with forced non-floating bound: 585808 (73.28%)
Realigned with forced non-floating right bound in left read: 314933 (39.39%)
Realigned with forced non-floating left bound in right read: 314933 (39.39%)
IGH chains: 175269 (23.74%)
IGH non-functional: 6761 (3.86%)
IGK chains: 255412 (34.59%)
IGK non-functional: 8651 (3.39%)
IGL chains: 307629 (41.67%)
IGL non-functional: 89916 (29.23%)
Trimming report:
  R1 reads trimmed left: 20 (0%)
  R1 reads trimmed right: 124220 (15.54%)
  Average R1 nucleotides trimmed left: 0.0028944537562854925
  Average R1 nucleotides trimmed right: 2.4917056513146374
  R2 reads trimmed left: 43806 (5.48%)
  R2 reads trimmed right: 799407 (99.99%)
  Average R2 nucleotides trimmed left: 1.5104232857178597
  Average R2 nucleotides trimmed right: 141.1181397443274
Tag parsing report:
  Execution time: 0ns
  Total reads: 799460
  Matched reads: 799460 (100%)
  Projection +R1 +R2: 799460 (100%)
  For variant 0:
    For projection +R1 +R2:
      R1:Left position: 0
      R1:Right position: 300
      R2:Left position: 40
      R2:Right position: 300
      Variants: 0
      Cost: 0
      R1 length: 300
      R2 length: 260

====================== mixcr assemble ======================
Running:
mixcr assemble --report /rsrch6/scratch/lym_myl_rsch/plakra/SMARCA4/bcr/analysis/638_mBCR.assemble.report.txt --json-report /rsrch6/scratch/lym_myl_rsch/plakra/SMARCA4/bcr/analysis/638_mBCR.assemble.report.json /rsrch6/scratch/lym_myl_rsch/plakra/SMARCA4/bcr/analysis/638_mBCR.vdjca /rsrch6/scratch/lym_myl_rsch/plakra/SMARCA4/bcr/analysis/638_mBCR.clns
Initialization: progress unknown
Assembling initial clonotypes: 61.3%
Mapping low quality reads: 7.4%
Mapping low quality reads: 25.6%  ETA: 00:00:04
Mapping low quality reads: 51.6%  ETA: 00:00:03
Mapping low quality reads: 91.6%  ETA: 00:00:00
Clustering: 0.1%
Clustering: 54.4%  ETA: 00:00:01
Building clones: 68.2%  ETA: 00:00:02
===================== report: assemble =====================
Analysis time: 10.58s
Final clonotype count: 730
Reads used in clonotypes, percent of total: 107362 (13.43%)
Average number of reads per clonotype: 147.07
Reads dropped due to the lack of a clone sequence, percent of total: 576027 (72.05%)
Reads dropped due to a too short clonal sequence, percent of total: 0 (0%)
Reads dropped due to low quality, percent of total: 0 (0%)
Reads dropped due to failed mapping, percent of total: 45700 (5.72%)
Reads dropped with low quality clones, percent of total: 327 (0.04%)
Aligned reads processed: 157741
Reads used in clonotypes before clustering, percent of total: 111714 (13.97%)
Number of reads used as a core, percent of used: 88077 (78.84%)
Mapped low quality reads, percent of used: 23637 (21.16%)
Reads clustered in PCR error correction, percent of used: 4352 (3.9%)
Reads pre-clustered due to the similar VJC-lists, percent of used: 12 (0.01%)
Clonotypes dropped as low quality: 282
Clonotypes eliminated by PCR error correction: 2440
Clonotypes pre-clustered due to the similar VJC-lists: 4
Clones dropped in post filtering: 0 (0%)
Reads dropped in post filtering: 0.0 (0%)
Alignments filtered by tag prefix: 0 (0%)
IGH chains: 13 (1.78%)
IGH non-functional: 0 (0%)
IGK chains: 529 (72.47%)
IGK non-functional: 19 (3.59%)
IGL chains: 188 (25.75%)
IGL non-functional: 55 (29.26%)

====================== mixcr qc ======================
Running:
mixcr qc --print-to-stdout /rsrch6/scratch/lym_myl_rsch/plakra/SMARCA4/bcr/analysis/638_mBCR.clns /rsrch6/scratch/lym_myl_rsch/plakra/SMARCA4/bcr/analysis/638_mBCR.qc.txt /rsrch6/scratch/lym_myl_rsch/plakra/SMARCA4/bcr/analysis/638_mBCR.qc.json

  Successfully aligned reads:                     92.35% [OK]
  Off target (non TCR/IG) reads:                  5.78%  [OK]
  Reads with no V or J hits:                      1.84%  [OK]
  Overlapped paired-end reads:                    26.72% [ALERT]
  Reads used in clonotypes:                       13.43% [ALERT]
  Alignments that do not cover VDJRegion:         72.05% [ALERT]
  Alignments dropped due to low sequence quality: 0.0%   [OK]
  Clones dropped in post-filtering:               0.0%   [OK]
  Alignments dropped in clones post-filtering:    0.0%   [OK]

====================== mixcr exportClones ======================
Running:
mixcr exportClones /rsrch6/scratch/lym_myl_rsch/plakra/SMARCA4/bcr/analysis/638_mBCR.clns /rsrch6/scratch/lym_myl_rsch/plakra/SMARCA4/bcr/analysis/638_mBCR.clones.tsv
The following analysis parameters will be used:
splitByTagType: null
filterOutOfFrames: false
filterStops: false
chains: ALL
noHeader: false
fields:
- field: -cloneId
- field: -readCount
- field: -readFraction
- field: -targetSequences
- field: -targetQualities
- field: -vHitsWithScore
- field: -dHitsWithScore
- field: -jHitsWithScore
- field: -cHitsWithScore
- field: -vAlignments
- field: -dAlignments
- field: -jAlignments
- field: -cAlignments
- field: -allNFeaturesWithMinQuality
- field: -allAAFeatures
- field: -defaultAnchorPoints
splitFilesBy:
- geneLabel:ReliableChain
groupClonesBy: []
====================
Filtered 13 of 730 clones (98.22%).
Filtered 1710.0 of 107362.0 reads (98.41%).
Filtered 188 of 730 clones (74.25%).
Filtered 45791.0 of 107362.0 reads (57.35%).
Filtered 529 of 730 clones (27.53%).
Filtered 59861.0 of 107362.0 reads (44.24%).
Analysis finished successfully.


Should I be concerned about this low number of reads used in clonotypes, and if it can impact detection of all possible clonotypes?

Thank you so much

**Reply by mizraelson** [reply]
Did you run and check FASTQC report? The issue seems to be with the sequencing quality of R2.

**Reply by drosop** [reply]
I just checked FASTQC. You are right, R2 reads have low quality.
![fastqc-status-check-heatmap (1)](https://github.com/milaboratory/mixcr/assets/135168045/3bd7337a-89c0-481e-a55d-ce48059fa629)



Can I use this type of data? or we need to resequence? We really need this data urgently. Is there any workaround?

**Reply by mizraelson** [reply, marked as answer]
It depends on the experiment. From the report, I see that almost all R2 reads have been trimmed by 140nt. This means that, on average, half of the nucleotides in each R2 read has a quality below 10 (the default threshold). If you are only interested in the CDR3 sequence, you can add `--assemble-clonotypes-by CDR3`, and it should be fine, as the CDR3 sequence is covered by R1. If you are interested in the full VDJ region, you can try turning off the quality-based trimming by adding `-Malign.trimmingQualityThreshold=0`. Probably, if the number of reads in clones is still low, you will also need to lower the clonotype assembly quality threshold by adding: `-Massemble.cloneAssemblerParameters.badQualityThreshold=0`. However, by doing this, you might not be able to distinguish hypermutations from sequencing errors accurately and could introduce artificial diversity. Therefore, if you are looking for precise hypermutation analysis, I would recommend resequencing.

**Reply by drosop** [reply]
Thank you so much for the detailed explanation. I really appreciate the help.
